assets/app/view/game/buy_companies.rb

```diff
@@ -32,11 +33,34 @@ def render_companies
           [company.owner == @corporation.owner ? 0 : 1, company.value]
         end
 
-        companies.map do |company|
+        companies_to_buy = companies.map do |company|
+          unless company.owner == @corporation.owner || @show_other_players
```

Comment:
don't use unless with multiple conditions

Reply:
I've replaced this with the equivalent `if` statement